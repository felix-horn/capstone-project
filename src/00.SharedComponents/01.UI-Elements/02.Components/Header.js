import { useHistory } from 'react-router-dom'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import ArrowBackIcon from '@material-ui/icons/ArrowBack'
import MoreVertIcon from '@material-ui/icons/MoreVert'

Header.propTypes = {
  onClick: PropTypes.func,
  shopId: PropTypes.string,
  className: PropTypes.string,
}

export default function Header({ onClick, shopId, className }) {
  const history = useHistory()
  console.log({ history })
  return (
    <LayoutWrapper className={className}>
      <BackButton onClick={navigateBack} data-testid="back-button">
        <ArrowBackIcon />
      </BackButton>
      {onClick && <MenuButton onClick={onClick} data-testid="menu-button" />}
    </LayoutWrapper>
  )
  function navigateBack() {
    // history.replace(shopId ? { pathname: '/shop', state: { shopId } } : '/')
    history.goBack()
  }
}

const LayoutWrapper = styled.div`
  box-shadow: var(--light-box-shadow);
  height: 60px;
  max-width: 450px;
  background-color: var(--white);
  display: grid;
  place-items: center;
  grid-template-columns: 60px auto 60px;
`

const BackButton = styled.div`
  grid-column: 1;
  margin-top: 2px;
  color: var(--dark-gray);
`

const MenuButton = styled(MoreVertIcon)`
  grid-column: 3;
  color: var(--dark-gray);
`
