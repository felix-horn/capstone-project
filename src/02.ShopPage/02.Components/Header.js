import { NavLink } from 'react-router-dom'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import ArrowBackIcon from '@material-ui/icons/ArrowBack'
import MoreVertIcon from '@material-ui/icons/MoreVert'

Header.propTypes = {
  onClick: PropTypes.func.isRequired,
}

export default function Header({ onClick }) {
  return (
    <HeaderStyled>
      <BackButton exact to="/">
        <ArrowBackIcon />
      </BackButton>
      <MoreVertIconStyled onClick={onClick} />
    </HeaderStyled>
  )
}

const HeaderStyled = styled.div`
  height: 60px;
  max-width: 450px;
  background-color: var(--white);
  display: grid;
  place-items: center;
  grid-template-columns: 60px auto 60px;
  box-shadow: var(--light-box-shadow);
`

const BackButton = styled(NavLink)`
  margin-top: 2px;
  color: var(--dark-gray);
`

const MoreVertIconStyled = styled(MoreVertIcon)`
  grid-column: 3;
  color: var(--dark-gray);
`
