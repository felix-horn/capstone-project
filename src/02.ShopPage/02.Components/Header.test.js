import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import { BrowserRouter as Router } from 'react-router-dom'
import Header from './Header'

describe('Header', () => {
  it('renders correctly', () => {
    const onClickMock = jest.fn()
    const { container } = render(
      <Router>
        <Header onClick={onClickMock} />
      </Router>
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('calls function onClick', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <Router>
        <Header onClick={onClickMock} />
      </Router>
    )
    const menuButton = getByTestId('menu-button')
    user.click(menuButton)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })

  it('links to correct href', () => {
    const onClickMock = jest.fn()
    const { getByTestId } = render(
      <Router>
        <Header onClick={onClickMock} />
      </Router>
    )
    const backButton = getByTestId('back-button')
    user.click(backButton)
    expect(backButton).toHaveProperty('href', 'http://localhost/')
  })
})
