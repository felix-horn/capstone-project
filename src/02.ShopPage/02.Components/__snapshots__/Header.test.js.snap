// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c0 {
  height: 60px;
  max-width: 450px;
  background-color: var(--white);
  display: grid;
  place-items: center;
  grid-template-columns: 60px auto 60px;
  box-shadow: var(--light-box-shadow);
}

.c1 {
  grid-column: 1;
  margin-top: 2px;
  color: var(--dark-gray);
}

.c2 {
  grid-column: 3;
  color: var(--dark-gray);
}

<div
  class="c0"
>
  <a
    aria-current="page"
    class="c1 active"
    data-testid="back-button"
    href="/"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
  </a>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root c2"
    data-testid="menu-button"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
    />
  </svg>
</div>
`;
